CONTRIBUTING.rst

```diff
@@ -264,6 +264,19 @@ This is also useful for writing unit tests.
 Writing small functions is not always possible, and we do not recommend trying to reorganize larger,
 but well-tested, older functions in the codebase, unless there is a strong reason to do so (e.g., when adding a new feature).
 
+.. admonition:: Recommendation
+
+    To lint your code and verify PEP8 compliance, you can run
+    `flake8 <https://flake8.pycqa.org/en/latest/>`_ locally on the
+    changes you have made in your branch compared to the main branch or a
+    previous commit.
+    To do this before committing changes, get the diff between the working
+    directory and the last commit and pipe it to flake8 by running:
+
+    .. code-block:: bash
+
+        git diff HEAD | flake8 --diff
```

Comment:
this will check the diff with HEAD, but we probably want the diff with the latest common ancestor with upstream/main, as is done here: https://github.com/nilearn/nilearn/blob/bd28810b2e6d97ef123d4c935c605ee384f0c40b/build_tools/flake8_diff.sh#L57

Reply:
Ok I think something like this should work when the branch is checked out:

```bash
COMMIT=$(git merge-base main @) 
git diff $COMMIT | flake8 --diff
```
This way working directory changes as well as all committed changes in the branch are checked. WDYT?

Reply:
I think it looks good!